### PR TITLE
Enable agent diagnostics reporting in Helm and yaml

### PR DESCRIPTION
**Pull request by alexgervais**
Signed-off-by: alex <alex@datawire.io>

## Description
This PR contains changes to:
- Enable agent diagnostics reporting in both Helm and standard yaml files.
- Serve diagnostics requests to `:8877/ambassador/v0/diag/`, even if the synthetics diagnostics mappings are disabled.
- Changelog entries for Helm and Emissary-ingress.

## Related Issues
None

## Testing
Manual tests

## Checklist

 - [x] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [ ] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [x] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -80,6 +80,17 @@ it will be removed; but as it won't be user-visible this isn't considered a brea
 - Feature: The agent is now able to parse api contracts using swagger 2, and to convert them to
   OpenAPI 3, making them available for use in the dev portal.
 
+- Feature: The agent now has the ability to report diagnostics information to Ambassador Cloud. If
+  you previous connected your installation to Ambassador Cloud, you may chose to enable this
```
Comment (LanceEa) [inline CHANGELOG.md:84]: nit: `previously`

File: manifests/emissary/emissary-emissaryns.yaml.in [hunk 2]
```diff
@@ -368,6 +368,7 @@ metadata:
 spec:
   config:
     diagnostics:
+      allow_non_local: true
```
[the remark this answers is not in the document]
Comment (alexgervais) [reply manifests/emissary/emissary-emissaryns.yaml.in:371]: Exactly! I was going to implement this feature when I started to read the code and stumbled upon this comment in `diagd.py`


> Allow non local traffic even when diagnotics UI is disabled. Mappings are not added for the diag UI but the diagnotics UI is still exposed for the pod IP in the admin port.

**Review by ddymko — APPROVED** [review]
looks good

File: manifests/emissary/emissary-emissaryns.yaml.in (hunk 2 above)
[the remark this answers is not in the document]
Comment (LanceEa) [reply manifests/emissary/emissary-emissaryns.yaml.in:371]: Ok thanks, I double checked and adding that flag does NOT add the routes for the admin so I think it is working as expected. 😄

**Pull request merged**